contracts/src/L1/L1BridgeController.sol

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.13;
+
+import {EjectionController} from "../common/EjectionController.sol";
+import {TransferData} from "../common/TransferData.sol";
+import {IRegistry} from "../common/IRegistry.sol";
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {LibRegistryRoles} from "../common/LibRegistryRoles.sol";
+import {IPermissionedRegistry} from "../common/IPermissionedRegistry.sol";
+import {IBridge, LibBridgeRoles} from "../common/IBridge.sol";
+import {BridgeEncoder} from "../common/BridgeEncoder.sol";
+import {NameUtils} from "../common/NameUtils.sol";
+import {NameCoder} from "@ens/contracts/utils/NameCoder.sol";
+
+/**
+ * @title L1BridgeController
+ * @dev L1 contract for bridge controller that facilitates migrations of names
+ * between L1 and L2, as well as handling renewals.
+ */
+contract L1BridgeController is EjectionController {
+    error NotTokenOwner(uint256 tokenId);
+    error NameNotExpired(uint256 tokenId, uint64 expires);
+    error ParentNotMigrated(bytes dnsEncodedName, uint256 offset);
```

Comment:
I think elsewhere we just use `name`.

Reply:
In some of this places the name is just the label and not the dnsEncodedName, so I wanted to be really specific here.